vault/client.go

```diff
@@ -0,0 +1,77 @@
+package vault
+
+import (
+	"errors"
+
+	vaultapi "github.com/hashicorp/vault/api"
+)
+
+// VaultClient Used to read and write secrets from vault
+type VaultClient struct {
+	token  string
+	client *vaultapi.Client
+}
+
+// CreateVaultClient by providing a auth token, vault address and the maxium number of retries for a request
+func CreateVaultClient(token, vaultAddress string, retries int) (*VaultClient, error) {
+	config := vaultapi.Config{Address: vaultAddress, MaxRetries: retries}
+	client, err := vaultapi.NewClient(&config)
+	if err != nil {
+		return nil, err
+	}
+	client.SetToken(token)
+	return &VaultClient{token, client}, nil
+}
+
+// CreateTLSVaultClient by providing a auth token, vault address and the maxium number of retries for a request
```

Comment:
this name irks me - does `CreateVaultClientTLS` sound better to you?

Reply:
same typos as above

Reply:
I renamed the function and fixed the typo